src/Agent/NewRelic/Agent/Core/AgentInstallConfiguration.cs

```diff
@@ -91,6 +91,13 @@ static AgentInstallConfiguration()
             AgentInfo = GetAgentInfo();
         }
 
+        //This method and delegate is here to allow dependency injecting the IsWindows logic via a typed delegate
+        public delegate bool IsWindowsDelegate();
```

Comment:
Does it make sense to invest in solving this a different way? This pattern of using a delegate seems to be a shortcut to avoid the work of making this a non-static class and introducing an interface. If we run with the delegate here for now, I'd like us to consider coming back soon and doing this differently.

Another thought that may be less effort than refactoring this entire class at the moment would be to just hang `IsWindows` off of `IConfiguration`.

Reply:
`IConfiguration` doesn't feel like a good workaround either, because it's not really a configuration. I considered making an interface and then only exposing IsWindows from it, and then using the same pattern that we are using for the Process class. 

However, all of this seems overkill for what we are trying to do. I really just want to dependency inject a boolean value into the .ctor, but I did not see a good way to handle that with the dependency injection containers that we are using. Part of me would rather just take the time to remove our reliance on these dependency injection containers because of all of the subtle and not-so-subtle we've run into with things not getting created or started when we expected them to. If we do that, then we could simply pass in the `AgentInstallConfiguration.IsWindows` value directly into the .ctor.

Reply:
True, I've always been a fan of eliminating the overly complex auto dependency injection we're using. Also, I do agree trying to deal with this at the moment is overkill for the task at hand. My primary concern is the propagating this pattern of using a delegate to other things further complicating something that should be far easier.

I'm ok moving forward with this solution for now. Perhaps we should open an issue for the removal of the DI framework. It has led to a stupid amount of issues over the years.